R/dependecies.R

```diff
@@ -0,0 +1,510 @@
+    filenames <- paste(url, strsplit(filenames, "\r*\n")[[1]], sep = "")
+    filenames <- filenames[grepl(paste0(name, "_[0-9\\.\\-]+\\.tar\\.gz"),filenames)]
+    expr <- gregexpr(paste0(name, "_([0-9\\.\\-]+)\\.tar\\.gz"), filenames)
+    filenames <- vapply(regmatches(filenames, expr), function(match) match[1], character(1))
+    archived_version <- gsub("(.*_)([0-9\\.\\-]+)(\\.tar\\.gz)", "\\2", filenames[length(filenames)])
+    if (is.na(version) || package_version(archived_version) > package_version(as.character(version))) {
+      version <- archived_version
+    }
+  } else{
+    # Simply download latest if version happens to be the latest available on CRAN.
+    remote_version <- as.character(remote_version)
+    if (is.na(version) || package_version(remote_version) == package_version(version)) {
+      version <- remote_version
+      archive_addition <- ""
+    } else{
+      archive_addition <- paste0("Archive/", name, "/")
+    }
+  }
+
+  from <- paste0(repo, "/src/contrib/", archive_addition, name, "_", version, ".tar.gz")
+  pkg_tarball <- tempfile(fileext = ".tar.gz")
+  out <- suppressWarnings(tryCatch(
+    download.file(url = from, destfile = pkg_tarball, quiet = notTRUE(getOption('lockbox.verbose')))
+    , error = function(e) e))
+  # Sometimes the current version isn't accessible in it's usual place, but is already archived
+  if (is(out, "error")) {
+    archive_addition <- paste0("Archive/", name, "/")
+    from <- paste0(repo, "/src/contrib/", archive_addition, name, "_", version, ".tar.gz")
+    download.file(url = from, destfile = pkg_tarball, quiet = notTRUE(getOption('lockbox.verbose')))
+  }
+  pkg_tarball
+}
+
+#' Download a package from github using devtools' remote_download function
+download_package.github <- function(package) {
+  if (is.na(package$version)) {
+    package$version <- NULL
+  }
+  remote <- get_remote(package)
+  quiet <- !isTRUE(getOption('lockbox.verbose'))
+  devtools:::remote_download.github_remote(remote, quiet = quiet)
+}
+
+#' Return the latest available version of a package from CRAN
+get_available_cran_version <- function(package, repo = "http://cran.r-project.org") {
+  repo = "http://cran.r-project.org"
+  # List available packages on the repo
+  available <- available.packages(contriburl =
+    contrib.url(repos = "http://cran.us.r-project.org", type = "source"))
+  available <- data.frame(unique(available[, c("Package", "Version")]))
+  pkg <- available[available$Package == package$name, ]
+  if (nrow(pkg) == 0) {
+    NA
+  } else{
+    pkg$Version
+  }
+}
+
+#' Create remote in form devtools' remote_download likes.
+get_remote <- function(package) {
+  ref <- package$ref %||% package$version
+  if (is.null(ref)) {
+    arguments <- list(package$repo)
+  } else {
+    arguments <- list(
+      paste(package$repo, ref, sep = "@"))
+  }
+  if (nzchar(token <- Sys.getenv("GITHUB_PAT"))) {
+    arguments$auth_token <- token
+  }
```

Comment:
Don't think you need the `if` protection here; I think it's okay for `arguments$auth_token` to be `""`.